packages/medusa/src/services/order-edit.ts

```diff
@@ -389,6 +392,67 @@ export default class OrderEditService extends TransactionBaseService {
     })
   }
 
+  async removeLineItem(
+    orderEditId: string,
+    lineItemId: string
+  ): Promise<OrderEdit> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEdit = await this.retrieve(orderEditId, {
+        select: [
+          "id",
+          "created_at",
+          "requested_at",
+          "confirmed_at",
+          "declined_at",
+          "canceled_at",
+        ],
+        relations: ["items"],
+      })
+
+      const isOrderEditActive = OrderEditService.isOrderEditActive(orderEdit)
+
+      if (!isOrderEditActive) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          `Can not update an item on the order edit ${orderEditId} with the status ${orderEdit.status}`
+        )
+      }
+
+      const deleteLineItem = orderEdit.items.find(
+        (it) => it.original_item_id === lineItemId
+      )
+
+      if (typeof deleteLineItem === "undefined") {
```

Comment:
todo: not necessary with the previous change

Reply:
would still have to check that `retrieve` doesn't throw an error so that the method can remain idempotent